frontend/src/main/components/Courses/CoursesForm.js

```diff
@@ -0,0 +1,168 @@
+                            <Form.Label htmlFor="id">Id</Form.Label>
+                            <Form.Control
+                                data-testid="CoursesForm-id"
+                                id="id"
+                                type="text"
+                                {...register("id")}
+                                value={initialContents.id}
+                                disabled
+                            />
+                        </Form.Group>
+                    </Col>
+                )}
+
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="name">Name</Form.Label>
+                        <Form.Control
+                            data-testid="CoursesForm-name"
+                            id="name"
+                            type="text"
+                            isInvalid={Boolean(errors.name)}
+                            {...register("name", { required: true })}
+                        />
+                        <Form.Control.Feedback type="invalid">
+                            {errors.name && 'Name is required. '}
+                        </Form.Control.Feedback>
+                    </Form.Group>
+                </Col>
+            </Row>
+
+            <Row>
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="school">School</Form.Label>
+                        <Form.Control
+                            data-testid="CoursesForm-school"
+                            id="school"
+                            type="text"
+                            isInvalid={Boolean(errors.school)}
+                            {...register("school", { required: true })}
+                        />
+                        <Form.Control.Feedback type="invalid">
+                            {errors.school && 'School is required. '}
+                        </Form.Control.Feedback>
+                    </Form.Group>
+                </Col>
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="term">Term</Form.Label>
+                        <Form.Control
+                            data-testid="CoursesForm-term"
+                            id="term"
+                            type="text"
+                            isInvalid={Boolean(errors.term)}
+                            {...register("term", { required: true })}
+                        />
+                        <Form.Control.Feedback type="invalid">
+                            {errors.term && 'Term is required. '}
+                        </Form.Control.Feedback>
+                    </Form.Group>
+                </Col>
+            </Row>
+
+            <Row>
+                <Col>
+                    <Form.Group className="mb-3" >
+                        <Form.Label htmlFor="start">Start(iso format)</Form.Label>
```

Comment:
do you need to specify iso format when using a date picker?

Reply:
team03 followed a similar example so I went with that. I don't think so though.

Reply:
It's true: you don't really need this.  I've asked other teams to take it out, but for sake of moving things along, I'll go ahead and let it slide.  But you might want to do a 5 pt refactoring PR later to remove those checks.